### ISSUE TITLE
Image with JS

### ISSUE THREAD

**Issue by Tymek** [pahen/madge#169]
It shouldn't be very hard to use `D3` for SVG rendering. PR welcome? It's been some time since I did something in D3, so no promises :)

**Comment by lll000111** [comment]
I think it's easiest to just use a graphviz based JS package like

- https://github.com/mdaines/viz.js (if that works)
- https://github.com/magjac/d3-graphviz
- and more (Google finds a few)

because you already have graphviz support. Otherwoise it's probably *a lot* of work since I think the package right now produces dot syntax instead of doing any drawing? So you would have to actually code the drawing all from scratch.

**Comment by shershen08** [comment]
@Tymek  Can d3-based render eliminate the need to separately install graphViz ?

**Comment by Tymek** [comment]
@shershen08 It will. @lll000111 good point, thanks! I managed to render SVGs with [Viz.js](https://github.com/mdaines/viz.js). Work in progress. After some reading, I think I'd go with [Dagre](https://github.com/dagrejs). Important thing to be decided is, should we drop GraphViz? I guess we can. If DOT export would still be provided.

TL;DR
Yeap. Workin' on it.

**Comment by Tymek** [comment]
@pahen should I continue? I don't want to code new feature just for it to be rejected.

**Comment by pahen** [comment]
Thanks for your effort here @Tymek! I have plans to support an interactive mode in Madge that doesn't depend on Graphviz but I will still support Graphviz for generating images. I'll dig into your findings when it's time to start working on it. Closing this for now.
